catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -355,9 +358,42 @@ private String getExceptionMessage(String responseType, String previousException
         return builder.toString();
     }
 
+    public void shutdown() {
+        getPeriodicDrain().shutdown();
```

Comment:
The article raises a couple of good points. One question though. If we do have a bunch of metacards left to process, shouldn't we make sure they all get taken care of before shutting down? Otherwise the backup will be incomplete as we'll be missing some of those metacards. Which pretty much rules out using `shutdownNow()`, and forces us to use `awaitForTermination()` as much as we can. We should probably log a real big error if we have to shutdown and don't know if all the metacards have been backed up.

I guess what's tricky here is that we're actually trading performance for completeness. If the app or bundle crashes or we don't shutdown cleanly, i.e., make sure all metacards have been processed, we could lose some metacards. Is this acceptable?